setup.py

```diff
@@ -16,29 +16,36 @@
 #   TORCH_XLA_PACKAGE_NAME
 #     change the package name to something other than 'torch_xla'
 #
-#   COMPILE_PARALLEL=1
-#     enable parallel compile
-#
-#   BUILD_CPP_TESTS=1
-#     build the C++ tests
-#
-#   XLA_DEBUG=0
-#     build the xla/xrt client in debug mode
```

Comment:
We need a way to control the debug mode. Is this always on?

Reply:
There was a difference before for DEBUG and XLA_DEBUG. Now only DEBUG is relevant since it's all a single binary, and that will be applied. Is that enough?

Reply:
Hey @stgpetrovic in your experince how reliable is the TF's debug build? In my experience it almost never worked hence I usually have `DEBUG=1` and `XLA_DEBUG=0` so at least I get the stack trace info for torch_xla.

Reply:
Hm, it fails to build in debug mode currently. We need to fix that. Let me look into it. We can't have the codebase in a state where `#ifdef NDEBUG` causes issues.

Reply:
There are some more issues with `dbg` builds :S

Not sure what to do here. I'll keep digging.

Reply:
Thanks Goran, that has been my experience :D. I fix one debug build error and then there is next, I felt like debug build is either not used or used differently internally so people break it pretty often without noticing.

Reply:
I managed to get it to work and I think it consistently will.

Basically, we can control per-file `copts` with bazel, so I told it to compile all non-pytorch/xla files with `-g0 -DNDEBUG`, which makes our whole repo (including the stuff in xla_client) have debug symbols, and the external stuff (tf, llvm, mlir) not. We can in the future maybe play with this to enable more debug symbols form TF in TF/OpenXLA core, but not the underlying deps that usually are the root cause of issues when compiling in debug mode.

This is a nice compromse IMHO.